File: 605.can-place-flowers.py
#
# @lc app=leetcode id=605 lang=python
#
# [605] Can Place Flowers
#

# @lc code=start
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        for i, x in enumerate(flowerbed):
            if not x and (i == 0 or flowerbed[i-1] == 0) \
                and (i == len(flowerbed)-1 or flowerbed[i+1] == 0):
                n -= 1
                flowerbed[i] = 1
        return n <= 0
# @lc code=end

File: 605.can-place-flowers.cpp
/*
 * @lc app=leetcode id=605 lang=cpp
 *
 * [605] Can Place Flowers
 */
#include <vector>
using namespace std;

// @lc code=start
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int length = flowerbed.size();
        int count = 0;
        for (int i = 0; i < length & count < n; i++) {
            if (flowerbed[i] == 1)
                continue;
            int pre = i == 0 ? 0 : flowerbed[i - 1];
            int next = i == length - 1 ? 0 : flowerbed[i + 1];
            if (pre == 0 && next==0) {
                count++;
                flowerbed[i] = 1;
            }
        }
        return count >= n;
    }
};
// @lc code=end

